runtime/jcl/common/thread.cpp

```diff
@@ -216,18 +224,24 @@ Java_java_lang_Thread_suspendImpl(JNIEnv *env, jobject rcv)
 	Trc_JCL_threadSuspend(currentThread, targetThread);
 	if (J9VMJAVALANGTHREAD_STARTED(currentThread, receiverObject)) {
 		if (NULL != targetThread) {
-			if (currentThread == targetThread) {
+#if JAVA_SPEC_VERSION >= 19
+			J9OBJECT_U32_STORE(currentThread, receiverObject, vm->isSuspendedInternalOffset, 1);
+			if (receiverObject == targetThread->threadObject)
+#endif /* JAVA_SPEC_VERSION >= 19 */
+			{
+				if (currentThread == targetThread) {
 				/* Suspending the current thread will take place upon re-entering the VM after returning from
 				 * this native.
 				 */
-				vmFuncs->setHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
-			} else {
-				vmFuncs->internalExitVMToJNI(currentThread);
-				omrthread_monitor_enter(targetThread->publicFlagsMutex);
-				VM_VMAccess::setHaltFlagForVMAccessRelease(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
-				if (VM_VMAccess::mustWaitForVMAccessRelease(targetThread)) {
-					while (J9_ARE_ALL_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND | J9_PUBLIC_FLAGS_VM_ACCESS)) {
-						omrthread_monitor_wait(targetThread->publicFlagsMutex);
+					vmFuncs->setHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
+				} else {
```

Comment:
During the review, I was looking at the code with white-space hidden. So, I missed that the `else` block is indented incorrectly. It needs to include the monitor exit and goto calls.

@dipak-bagadiya This might be the potential fix for the failure seen in https://github.com/eclipse-openj9/openj9/issues/17334.

fyi @pshipton, we will thoroughly test the code before merging again.

**Revised code:**
```
...
#if JAVA_SPEC_VERSION >= 19
J9OBJECT_U32_STORE(currentThread, receiverObject, vm->isSuspendedInternalOffset, 1);
if (receiverObject == targetThread->threadObject)
#endif /* JAVA_SPEC_VERSION >= 19 */
{
	if (currentThread == targetThread) {
		vmFuncs->setHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
	} else {
		vmFuncs->internalExitVMToJNI(currentThread);
		omrthread_monitor_enter(targetThread->publicFlagsMutex);
		VM_VMAccess::setHaltFlagForVMAccessRelease(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
		if (VM_VMAccess::mustWaitForVMAccessRelease(targetThread)) {
			while (J9_ARE_ALL_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND | J9_PUBLIC_FLAGS_VM_ACCESS)) {
				omrthread_monitor_wait(targetThread->publicFlagsMutex);
			}
		}
		omrthread_monitor_exit(targetThread->publicFlagsMutex);
		goto vmAccessReleased;
	}
}
...
```